03-task-manager/starter/controllers/tasks.js

```diff
@@ -1,33 +1,78 @@
 const Task = require('../models/Task')
-const getALLTasks = ( req,res) => {
-    res.send('Get all task')
-}
+const asyncWrapper = require('../middleware/async')
+const {createCustomError} = require('../errors/custom-error')
 
+const getALLTasks = asyncWrapper (async (req,res) => {
+ const tasks = await Task.find({})
+res.status(200).json({tasks})
+    
+    })
 
 
-const createTask = async (req, res) => {
-    const task = await Task.create(req.body)
-    res.status(201).json({task})
-}
 
-const getTask = (req, res) => {
-    res.json({id: req.params.id})
-}
 
-const updateTask = (req, res) => {
-    res.send('Update task')
-}
+const createTask = asyncWrapper (async (req, res) => {
+   
+        const task = await Task.create(req.body)
+        res.status(201).json({task}) 
+   
+    
+})
+
+const getTask = asyncWrapper (async (req, res, next) => {
```

Comment:
Nitpick:
Within your **controllers**, I'd recommend changing the `next` parameter to `done` within all of your controllers that need it. It's a common pattern that says "Hey, this callback is the last function to be called".